core/src/main/java/org/mskcc/cbio/portal/dao/DaoCancerStudy.java

```diff
@@ -319,6 +319,13 @@ public static void addCancerStudy(CancerStudy cancerStudy, boolean overwrite) th
             //TODO - use this field in parts of the system that build up the list of studies to display in home page:
             pstmt.setInt(10, Status.UNAVAILABLE.ordinal());
             pstmt.setDate(11, java.sql.Date.valueOf(LocalDate.now()));
+            try {
+                ReferenceGenome referenceGenome = DaoReferenceGenome.getReferenceGenomeByGenomeName(cancerStudy.getReferenceGenome());
+                pstmt.setInt(12, referenceGenome.getReferenceGenomeId());
+            }
+            catch (NullPointerException e) {
+                pstmt.setInt(12,1); //#TODO default reference genome to use
```

Comment:
good point! Agree, Grep reference genome information portal properties file is the best way to go.

> species and genomic information
species=human
ncbi.build=37
ucsc.build=hg19

Reply:
does seems there is an easy way to read portal properties from a Dao object class. Any suggestions? 

> perhaps rather than hardcoding a "1" here, we should read a property from portal.properties with the name of a default reference genome to use when creating a study which specifies an unrecognized reference genome (or does not specify a reference genome).

Reply:
@sheridancbio , are we allow to read portal.properties from a Dao object class?

Reply:
@sheridancbio @ao508 , if you take look at the code block (line 108~114) in CancerStudyReader.java 
if no **_reference genome_** info included in the meta_study.txt file, CancerStudyReader actually reads from the portal.properties file.
>         String referenceGenome = properties.getProperty("reference_genome");
 So, I will remove hard coded 1 with a SQL exception to indicate an unsupported reference genome.